### PR TITLE
feat(remix-vercel): show deprecation warning on usage

### PR DESCRIPTION
**Pull request by MichaelDeBoey**
As @mcansh mentioned in https://github.com/remix-run/remix/pull/5734#issuecomment-1478469486

### CODE REVIEW

**Comment by mcansh** [comment]
we're still deciding on what to do with our vercel adapter at the moment, so going to hold off on this one for now

**Review by mcansh — COMMENTED** [review]
instead of warning on usage, can we add this message to the top of the vercel template readme?

**Comment by MichaelDeBoey** [comment]
@mcansh We could do both
What I had also in mind was adding the deprecation warning to the npm package itself

File: packages/remix-vercel/index.ts [hunk 1]
```diff
@@ -1,4 +1,22 @@
...
+  }
+};
+
+warnOnce(
+  "⚠️ DEPRECATED: The `@remix-run/vercel` runtime adapter has been " +
+    "deprecated in favor of out of the box Vercel functionality and will be " +
+    "removed in Remix v2. Please update your code by removing " +
+    "`@remix-run/vercel` & `@vercel/node` from your `package.json`, removing " +
+    "your `server.js`/`server.ts` file, and removing the `server` & " +
+    "`serverBuildPath` options from your `remix.config.json`.",
```
Comment (brophdawg11) [inline packages/remix-vercel/index.ts:17]:
```suggestion
    "`serverBuildPath` options from your `remix.config.js`.",
```

**Review by brophdawg11 — APPROVED** [review]
Confirmed with @mjackson we were good to deprecate this entirely in v2

File: packages/remix-vercel/index.ts [hunk 2]
```diff
@@ -1,4 +1,22 @@
...
+const alreadyWarned: Record<string, boolean> = {};
+const warnOnce = (message: string, key = message) => {
+  if (!alreadyWarned[key]) {
+    alreadyWarned[key] = true;
+    console.warn(message);
+  }
+};
+
+warnOnce(
+  "⚠️ DEPRECATED: The `@remix-run/vercel` runtime adapter has been " +
```
[the remark this answers is not in the document]
Comment (MichaelDeBoey) [reply packages/remix-vercel/index.ts:12]: This PR has been waiting for some time now 🙈

**Comment by brophdawg11** [comment]
Hm, we must have a bug in our bot logic - this is not included in `1.17.1-pre.1` and should land in `1.18.0`

https://github.com/remix-run/remix/compare/remix@1.17.0...remix@1.17.1-pre.1

**Comment by mcansh** [comment]
> Hm, we must have a bug in our bot logic - this is not included in `1.17.1-pre.1` and should land in `1.18.0`
> 
> 
> 
> https://github.com/remix-run/remix/compare/remix@1.17.0...remix@1.17.1-pre.1

yeah that’s odd… it gets the commits via `git log --pretty=format:%H v0.0.0-nightly-eb06147-20230614...remix@1.17.1-pre.1 ./packages`, but seems like github actions got a different list 🤔

**Comment by TrySound** [comment]
Seems like with vercel maintaining a fork of remix we getting less usable integration. As every remix bump is blocked until vercel decides to rebase their fork.

image


image

**Pull request merged**